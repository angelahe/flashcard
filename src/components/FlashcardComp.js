import React from 'react';
import NavBarComp from './NavBarComp';
import ManageDecksComp from './ManageDecksComp';
import FlashcardDecksComp from './flashcard/FlashcardDecksComp';
import FlashcardChartComp from './FlashcardChartComp';
import FlashcardSettingsComp from './FlashcardSettingsComp';
import FlashcardInfoComp from './FlashcardInfoComp';

class FlashcardComp extends React.Component {
  constructor() {
    super();
    this.state = {
      currentView: 'ManageDecks',
    };
  }

  setView = (newView = 'ManageDecks') => {
    switch (newView) {
      case 'ManageDecks':
      case 'FlashcardChart':
      case 'FlashcardSettings':
      case 'FlashcardInfo':
      case 'FlashcardDecks':
        this.setState({ currentView: newView });
        break;
      default:
        this.setState({ currentView: 'ManageDecks' });
    }
  }

  handleNavClick = (navClick) => {
    this.setView(navClick);
  }

  handleAddClick = () => {
    const { currentView } = this.state;
    switch (currentView) {
      case 'FlashcardDecks': {
        this.setView('ManageDecks');
        break;
      }
      default:
        this.setView('ManageDecks');
    }
  };

  render() {
    const { currentView } = this.state;
    return (
      <div className="FlashcardComp">
        <div className="WorkArea">
          { (currentView === 'ManageDecks')
            ? <ManageDecksComp />
            : null
          }
          { (currentView === 'FlashcardDecks')
            ? <FlashcardDecksComp onAdd={this.handleAddClick} />
            : null
          }
          { (currentView === 'FlashcardChart')
            ? <FlashcardChartComp />
            : null
          }
          { (currentView === 'FlashcardSettings')
            ? <FlashcardSettingsComp />
            : null
          }
          { (currentView === 'FlashcardInfo')
            ? <FlashcardInfoComp />
            : null
          }
        </div>
        <NavBarComp onNavClick={this.handleNavClick} />
      </div>
    );
  }
}
export default FlashcardComp;
