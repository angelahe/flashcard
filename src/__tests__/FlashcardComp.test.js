import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import FlashcardComp from '../components/FlashcardComp';

it('renders without crashing', () => {
  fetch.once(JSON.stringify([{ deck_id: '76ff91c7-8c30-4226-93f2-2c295fdb9939' }, { deck_id: '1a614b6b-408a-4365-94be-2b87be6b94dc' }]));
  shallow(<FlashcardComp />);
});

it('renders without crashing', () => {
  fetch.once(JSON.stringify([{ deck_id: '76ff91c7-8c30-4226-93f2-2c295fdb9939' }, { deck_id: '1a614b6b-408a-4365-94be-2b87be6b94dc' }]));
  const div = document.createElement('div');
  ReactDOM.render(<FlashcardComp />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('FlashcardComp renders correctly', () => {
  fetch.once(JSON.stringify([{ deck_id: '76ff91c7-8c30-4226-93f2-2c295fdb9939' }, { deck_id: '1a614b6b-408a-4365-94be-2b87be6b94dc' }]));
  const tree = renderer.create(<FlashcardComp />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('contains the container div FlashcardComp on shallow render', () => {
  const component = shallow(<FlashcardComp />);
  expect(component.html()).toContain('FlashcardComp');
});

//  handleNavClick tests
it('shows ManageDecksComp', () => {
  const component = shallow(<FlashcardComp />);
  component.instance().handleNavClick('ManageDecks');
  expect(component.instance().state.currentView).toBe('ManageDecks');
  expect(component.html()).toContain('ManageDecksComp');
});

it('shows FlashcardDecksComp', () => {
  const component = shallow(<FlashcardComp />);
  component.instance().handleNavClick('FlashcardDecks');
  expect(component.instance().state.currentView).toBe('FlashcardDecks');
  expect(component.html()).toContain('FlashcardDecksComp');
});

it('shows FlashcardChartComp', () => {
  const component = shallow(<FlashcardComp />);
  component.instance().handleNavClick('FlashcardChart');
  expect(component.instance().state.currentView).toBe('FlashcardChart');
  expect(component.html()).toContain('FlashcardChartComp');
});

it('shows FlsahcardSettingsComp', () => {
  const component = shallow(<FlashcardComp />);
  component.instance().handleNavClick('FlashcardSettings');
  expect(component.instance().state.currentView).toBe('FlashcardSettings');
  expect(component.html()).toContain('FlashcardSettingsComp');
});

it('shows FlashcardInfoComp', () => {
  const component = shallow(<FlashcardComp />);
  component.instance().handleNavClick('FlashcardInfo');
  expect(component.instance().state.currentView).toBe('FlashcardInfo');
  expect(component.html()).toContain('FlashcardInfoComp');
});

it('shows ManageDecks on unknown view', () => {
  const component = shallow(<FlashcardComp />);
  component.instance().setView('JunkView');
  expect(component.instance().state.currentView).toBe('ManageDecks');
  expect(component.html()).toContain('ManageDecksComp');
});

it('shows ManageDecks on Add click from FlashcardDecksComp', () => {
  const component = shallow(<FlashcardComp />);
  component.instance().handleNavClick('FlashcardDecks');
  component.instance().handleAddClick();
  expect(component.instance().state.currentView).toBe('ManageDecks');
  expect(component.html()).toContain('ManageDecksComp');
});

it('shows ManageDecks on Add click by default e.g. from charts', () => {
  const component = shallow(<FlashcardComp />);
  component.instance().handleNavClick('FlashcardCharts');
  component.instance().handleAddClick();
  expect(component.instance().state.currentView).toBe('ManageDecks');
  expect(component.html()).toContain('ManageDecksComp');
});
