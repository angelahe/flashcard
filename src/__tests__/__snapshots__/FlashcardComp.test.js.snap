// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlashcardComp renders correctly 1`] = `
<div>
  <div
    className="WorkArea"
  >
    <div>
      <div
        className="ActionBar"
      >
        <img
          alt="logo"
          className="NavImg NavBtn"
          src="logo_FFFFFF.png"
        />
        <button
          className="NavBtn"
          onClick={[Function]}
          type="button"
        >
          <img
            alt="Add"
            className="NavImg"
            src="add_FFFFFF.png"
          />
        </button>
        <span
          className="HeaderText"
        >
          Decks
        </span>
        <button
          className="NavBtn"
          onClick={[Function]}
          type="button"
        >
          <img
            alt="Back"
            className="NavImg"
            src="back_FFFFFF.png"
          />
        </button>
      </div>
      <div>
        <div
          className="AppList"
        />
      </div>
    </div>
  </div>
  <div
    className="NavBar"
  >
    <button
      className="NavBtn"
      type="button"
    >
      <img
        alt="Add"
        className="NavImg"
        src="decks_FFFFFF.png"
      />
    </button>
    <button
      className="NavBtn"
      onClick={[Function]}
      type="button"
    >
      <img
        alt="Add"
        className="NavImg"
        src="listOff_FFFFFF.png"
      />
    </button>
    <button
      className="NavBtn"
      onClick={[Function]}
      type="button"
    >
      <img
        alt="Add"
        className="NavImg"
        src="chartOff_FFFFFF.png"
      />
    </button>
    <button
      className="NavBtn"
      onClick={[Function]}
      type="button"
    >
      <img
        alt="Add"
        className="NavImg"
        src="settingsOff_FFFFFF.png"
      />
    </button>
    <button
      className="NavBtn"
      onClick={[Function]}
      type="button"
    >
      <img
        alt="Add"
        className="NavImg"
        src="infoOff_FFFFFF.png"
      />
    </button>
  </div>
</div>
`;
